sphinxlint.py

```diff
@@ -196,124 +197,206 @@ def is_in_a_table(error, line):
         if backtick_in_front_of_role.search(line):
             yield lno, "superfluous backtick in front of role"
+
+
+@checker(".rst")
+def check_hyperlinks_missing_space(file, lines):
+    """Search for hyperlinks missing a space, like: `Link text<https://example.com>`."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
         for match in seems_hyperlink_re.finditer(line):
             if not match.group(1):
                 yield lno, "missing space before < in hyperlink"
+
+
+@checker(".rst")
+def check_hyperlinks_missing_final_underscore(file, lines):
+    """Search for hyperlinks missing underscore after their closing backtick."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
+        for match in seems_hyperlink_re.finditer(line):
             if not match.group(2):
                 yield lno, "missing underscore after closing backtick in hyperlink"
+
+
+@checker(".rst")
+def check_roles_with_double_backtick(file, lines):
+    """Search for roles with double backticks like: :fct:``sum``."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
         if double_backtick_role.search(line):
             yield lno, "role use a single backtick, double backtick found."
+
+
+@checker(".rst")
+def check_roles_missing_leading_space(file, lines):
+    """Search for spaces missing before roles, like the:fct:`sum`."""
+    for lno, line in enumerate(lines, start=1):
+        if "`" not in line:
+            continue
         if role_glued_with_word.search(line):
             yield lno, "missing space before role"
-        if role_missing_right_column.search(line):
-            yield lno, "role missing column before first backtick."
+
+
+@checker(".rst")
+def check_roles_missing_colon_before_backtick(file, lines):
+    """Search for roles missing a colon, like: :issue`123`."""
+    for lno, line in enumerate(lines, start=1):
+        if role_missing_right_colon.search(line):
+            yield lno, "role missing colon before first backtick."
 
 
 @checker(".py", ".rst", rst_only=False)
-def check_whitespace(file, lines):
-    """Check for whitespace and line length issues."""
-    lno = line = None
+def check_carriage_returns(file, lines):
+    """Check for carriage returns (\\r) in lines."""
     for lno, line in enumerate(lines):
         if "\r" in line:
             yield lno + 1, "\\r in line"
+
+
+@checker(".py", ".rst", rst_only=False)
+def check_horizontal_tabs(file, lines):
+    """Check for horizontal tabs (\\t) in lines."""
+    for lno, line in enumerate(lines):
         if "\t" in line:
             yield lno + 1, "OMG TABS!!!1"
+
+
+@checker(".py", ".rst", rst_only=False)
+def check_trailing_whitespaces(file, lines):
+    """Check for trailing whitespaces at end of lines."""
+    for lno, line in enumerate(lines):
         if line.rstrip("\n").rstrip(" \t") != line.rstrip("\n"):
```

Comment:
```suggestion
        stripped_line = line.rstrip("\n")
        if stripped_line.rstrip(" \t") != stripped_line:
```
Avoids double strip.